src/appInsights.ts

```diff
@@ -16,23 +16,28 @@ import { app } from './utils/app.js';
 const __dirname = url.fileURLToPath(new URL('.', import.meta.url));
 appInsights
   .setup('InstrumentationKey=6b908c80-d09f-4cf6-8274-e54349a0149a;IngestionEndpoint=https://westeurope-3.in.applicationinsights.azure.com/;LiveEndpoint=https://westeurope.livediagnostics.monitor.azure.com/;ApplicationId=d42020c6-da32-44f7-bbb7-666f0b3a664d')
+  .setAutoCollectRequests(false)
+  .setAutoCollectPerformance(false, false)
+  .setAutoCollectExceptions(false)
+  .setAutoCollectDependencies(false)
+  .setAutoCollectConsole(false, false)
+  .setAutoCollectPreAggregatedMetrics(false)
+  .setSendLiveMetrics(false)
+  .enableWebInstrumentation(false)
   .setInternalLogging(false, false);
 // append -dev to the version number when ran locally
 // to distinguish production and dev version of the CLI
 // in the telemetry
 const version: string = `${app.packageJson().version}${fs.existsSync(path.join(__dirname, `..${path.sep}src`)) ? '-dev' : ''}`;
 const env: string = process.env.CLIMICROSOFT365_ENV !== undefined ? process.env.CLIMICROSOFT365_ENV : '';
-const appInsightsClient: appInsights.TelemetryClient = ((appInsights as any).default as typeof appInsights).defaultClient;
+const appInsightsClient: appInsights.TelemetryClient = appInsights.defaultClient;
 appInsightsClient.commonProperties = {
   version: version,
   node: process.version,
   env: env,
   ci: Boolean(process.env.CI).toString()
 };
 
-appInsightsClient.config.proxyHttpUrl = process.env.HTTP_PROXY ?? '';
```

Comment:
Rather than configuring proxy directly on app insights, we let @azure/core-rest-pipeline pick it up itself from env vars, which circumvents the broken code in core-rest-pipeline.